plone/namedfile/image_exif_rotation.py

```diff
@@ -0,0 +1,123 @@
+from StringIO import StringIO
+import PIL.Image
+import exifread as exif
+
+
+FLIP_LEFT_RIGHT = 0
+FLIP_TOP_BOTTOM = 1
+ROTATE_90 = 2
+ROTATE_180 = 3
+ROTATE_270 = 4
+TRANSPOSE_MAP = {
+    FLIP_LEFT_RIGHT: (u'Flip around vertical axis'),
+    FLIP_TOP_BOTTOM: (u'Flip around horizontal axis'),
+    ROTATE_270: (u'Rotate 90 clockwise'),
+    ROTATE_180: (u'Rotate 180'),
+    ROTATE_90: (u'Rotate 90 counterclockwise'),
+
+}
+AUTO_ROTATE_MAP = {
+    0: None,
+    90: ROTATE_270,
+    180: ROTATE_180,
+    270: ROTATE_90,
+}
+MIRROR_ROTATE_MAP = {
+    0: None,
+    90: ROTATE_270,
+    180: ROTATE_180,
+    270: ROTATE_90,
+}
+
+ROTATION = {'Horizontal (normal)': 1,
+            'Mirrored horizontal': 2,
+            'Rotated 180': 3,
+            'Mirrored vertical': 4,
+            'Mirrored horizontal then rotated 90 CCW': 5,
+            'Rotated 90 CW': 6,
+            'Mirrored horizontal then rotated 90 CW': 7,
+            'Rotated 90 CCW': 8}
+
+
+def exif_rotation(self, event):
```

Comment:
Style here. Please don't use "self". "self" is meant to be used for class instance methods. These are functions and variable names should be "obj" or "content". This needs to be changed in all the functions.